lib/Cake/Test/Case/Utility/HashTest.php

```diff
@@ -837,6 +837,28 @@ public function testExtractStringKey() {
 	}
 
 /**
+ * Test wildcard matcher
+ *
+ * @return void
+ */
+	public function testExtractWildcard() {
+		$data = array(
```

Comment:
Your test case has only strings as key. The extract using `{s}` would cover that and will not justify the need of `{*}`. I would add different key types (ie, mixing numbers, strings, boolean, etc).